ArduPlane/GCS_Mavlink.cpp

```diff
@@ -967,10 +959,14 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_guided_slew_commands(const mavl
             plane.guided_state.target_alt_accel = fabsf(packet.param3);
         }
 
-         // assign an acceleration direction
-        if (plane.guided_state.target_alt < plane.current_loc.alt) {
+         // assign an acceleration direction - this only works if the current and target frame are the same
+         // this should not be happening here - it should be in ModeGuided::update_target_altitude()
```

Comment:
I agree, this is such a big rework now you might as well just fix it there. It also means it works with any alt frame and you don't need `last_target_alt_frame` anymore.